nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/sensitive/ProtectedNiFiProperties.java

```diff
@@ -510,22 +448,21 @@ private boolean isProviderAvailable(String protectionScheme) {
      */
     private String unprotectValue(String key, String retrievedValue) {
         // Checks if the key is sensitive and marked as protected
-        if (isPropertyProtected(key)) {
-            final String protectionScheme = getProperty(getProtectionKey(key));
+        if (!isPropertyProtected(key)) {
+            return retrievedValue;
+        }
+        final String protectionScheme = getProperty(getProtectionKey(key));
 
-            // No provider registered for this scheme, so just return the value
-            if (!isProviderAvailable(protectionScheme)) {
-                logger.warn("No provider available for {} so passing the protected {} value back", protectionScheme, key);
-                return retrievedValue;
-            }
+        if (protectionScheme.equals("unknown")) {
+            return retrievedValue;
+        }
 
-            try {
-                SensitivePropertyProvider sensitivePropertyProvider = getSensitivePropertyProvider(protectionScheme);
-                return sensitivePropertyProvider.unprotect(retrievedValue);
-            } catch (SensitivePropertyProtectionException e) {
-                throw new SensitivePropertyProtectionException("Error unprotecting value for " + key, e.getCause());
-            }
+        // try and make one to unprotect, and if that fails...
+        try {
+            return StandardSensitivePropertyProvider.fromKey(defaultKeyHex).unprotect(retrievedValue);
+        } catch (final SensitivePropertyProtectionException e) {
+            throw new SensitivePropertyProtectionException(e);
```

Comment:
I think the prior behavior of wrapping the cause in a generic exception for visibility safety was desired.

Reply:
Ah, that's another good catch.  Restored the previous exception (re-)handling.